src/c++/perf_analyzer/genai-perf/genai_perf/llm_metrics.py

```diff
@@ -370,6 +371,26 @@ def export_to_csv(self, csv_filename: str):
             for row in singular_metric_rows:
                 csv_writer.writerow(row)
 
+    def export_parquet(self, parquet_filename: str):
```

Comment:
This is a nuance of the stats class.
We need to be able to output it. 
The metrics class feels like the place to keep things small and modular but the statistics class should handle all statistics related work.

@dyastremsky what are your thoughts?

Reply:
Honestly, I think we just need a big refactor. Perhaps we all feel that to some level.

As far as this specifically, the logic seems tightly coupled with the Statistics class, so I don't see what the benefit would be of breaking it out is. I personally wouldn't prioritize this with our timeline for this feature. There is likely a better way to refactor these classes generally though. We'd probably want to look at best-in-class examples. Perhaps some sort of utility classes for Statistics, such as one for exporting, possibly another for formatting.

Reply:
In my mind, the single responsibility of the Statistics class is to take all the metrics data and aggregate them into statistics. And it seems like there is nothing in the `export_parquet` method that has anything to do with statistics. It's handling metrics data under the hood and no statistics data is involved. 

I'm just raising this because I think it's important to think about tech debt as we develop, but I also agree that this is not the biggest concern that we have right now so I'll leave the final call to @debermudez .

Reply:
I see. That's a good point, I had not realized this is metrics-specific here. It sounds like Statistics is now overloaded and doing the calculations, formatting, and exporting for everything related to metrics, which could probably be broken into separate classes in some way. Deferring to Elias as well, especially given the timeline. A thoughtful refactor sounds like a priority in the near term.